test/categorical_algebra/DPO.jl

```diff
@@ -0,0 +1,738 @@
+module TestDPO
+using Test
+using Catlab.Graphs
+using Catlab.Present
+using Catlab.WiringDiagrams
+using Catlab.CategoricalAlgebra.FinSets
+using Catlab, Catlab.Theories, Catlab.CategoricalAlgebra
+using Catlab.Theories: AttrDesc
+
+######################################################################################
+######################################################################################
+# Wiring diagrams
+######################################################################################
+######################################################################################
+A, B, C, D = [:A], [:B], [:C], [:D];
+f,g,h,i = Box(:f, A, B), Box(:g, B, C),Box(:h, C, D), Box(:i, [:D,:B], [:D,:B]);
+
+GWD = WiringDiagram([:A,:A],[:D,:B])
+fv, gv, hv, fv2 = add_box!(GWD, f), add_box!(GWD, g), add_box!(GWD, h), add_box!(GWD, f);
+add_wires!(GWD, Pair[
+  (input_id(GWD),1) => (fv,1),
+  (fv,1) => (gv,1),
+  (gv,1) => (hv,1),
+  (hv,1) => (output_id(GWD),1),
+  (input_id(GWD),2) => (fv2,1),
+  (fv2,1) => (output_id(GWD),2)]);
```

Comment:
Not necessary for this PR, but JSYK, the `@program` macro is convenient for avoiding all this boilerplate when constructing DWDs.